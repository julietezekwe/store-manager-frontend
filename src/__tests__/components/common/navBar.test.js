/* eslint-disable no-undef */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import NavBar from '../../../components/common/NavBar.jsx';


let wrapper;
const props = {
  user: {
    token: 'bhwqjkbj',
    role: 'attendant',
    logout: jest.fn(),
  },
};
describe('Menu Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<NavBar {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
  describe('Elements', () => {
    beforeEach(() => {
      wrapper = mount(<MemoryRouter><NavBar {...props}/></MemoryRouter>);
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should have one add to cart when attendant is logged in element', () => {
      expect(wrapper.find('.fa-shopping-cart').length).toEqual(1);
    });
    it('should have login link for non logged in users', () => {
      const newProps = {
        user: {
          token: '',
          role: '',
          logout: jest.fn(),
        },
      };
      wrapper = mount(<MemoryRouter><NavBar {...newProps}/></MemoryRouter>);
      expect(wrapper.find('#auth').first().text().trim()).toEqual('login');
    //   console.log(wrapper.find('#auth').at(0).text().length);
    });
  });
});
