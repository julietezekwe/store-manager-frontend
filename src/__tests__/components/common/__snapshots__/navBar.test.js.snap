// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component should match snapshot 1`] = `
<header
  id="header"
>
  <nav>
     
    <div
      className="logo"
    >
      <i
        className="fab fa-stripe-s"
      />
      tore 
      <i
        className="fab fa-medium-m"
      />
      anager
    </div>
    <ul>
      <li>
        <a
          href="/attendant/checkout"
        >
          <i
            className="fas fa-shopping-cart"
          />
          <i
            style={
              Object {
                "background": "brown",
                "borderRadius": "100%",
                "color": "white",
                "display": "inline-block",
                "fontSize": "10px",
                "height": "13px",
                "lineHeight": "13px",
                "textAlign": "center",
                "width": "13px",
              }
            }
          />
        </a>
      </li>
      <li>
        <i
          className="fas fa-user-lock"
        />
      </li>
      <li>
        <i
          className="fas fa-sign-in-alt"
        >
           
          <span>
             Logout
          </span>
        </i>
      </li>
    </ul>
  </nav>
</header>
`;
